demos/python_demos/common/performance_metrics.py

```diff
@@ -0,0 +1,96 @@
+"""
+ Copyright (C) 2020 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from time import perf_counter
+import cv2
+from helpers import put_highlighted_text
+
+
+class Statistic:
+    def __init__(self):
+        self.latency = 0.0
+        self.duration = 0.0
+        self.frame_count = 0
+
+
+class PerformanceMetrics:
+    def __init__(self, time_window=1.0):    # defines the length of the timespan used for 'current fps' calculation,
+        self.time_window_size = time_window # set to '1 second' by default
+        self.last_moving_statistic = Statistic()
+        self.current_moving_statistic = Statistic()
+        self.total_statistic = Statistic()
+    
+    def update(self, last_request_start_time, frame, position=(15, 30),
+               font_scale=0.75, color=(200, 10, 10), thickness=2):
+        current_time = perf_counter()
+        if not hasattr(self, 'last_update_time'):
+            self.last_update_time = current_time
+
+        last_request_latency = current_time - last_request_start_time
+        if self.current_moving_statistic.frame_count != 1 or self.total_statistic.frame_count != 0:
+            self.current_moving_statistic.latency += last_request_latency
+            self.current_moving_statistic.duration = current_time - self.last_update_time
+            self.current_moving_statistic.frame_count += 1
+        else:
+            # for 1st frame
+            self.last_moving_statistic.latency = last_request_latency
+            self.last_moving_statistic.frame_count = 1
+            self.total_statistic.latency = last_request_latency
+            self.total_statistic.frame_count = 1
+
+        if current_time - self.last_update_time > self.time_window_size:
```

Comment:
@AnthonyQuantum, this one is left

Reply:
> I think you should also check if new data was collected because if the pipeline works slower than `time_window_size` you will store 0 in `last_moving_statistic`.

Even when the pipeline works slower than `time_window_size`, there is at least one value in `last_moving_statistic`.
It's because every time `update()` is called, one value is collected: `last_request_latency = current_time - last_request_start_time`.

I checked it on smallest time_window sizes like 0.0 and 0.1 and it still has at least one value.

Reply:
Ok